vod/common.h

```diff
@@ -16,8 +16,15 @@
 #define vod_div_ceil(x, y) (((x) + (y) - 1) / (y))
 #define vod_array_entries(x) (sizeof(x) / sizeof(x[0]))
 
-#define vod_is_bit_set(mask, index) (((mask)[(index) >> 3] >> ((index) & 7)) & 1)
-#define vod_set_bit(mask, index) (mask)[(index) >> 3] |= 1 << ((index) & 7)
+// bit sets
+#define vod_number_of_bits(i) (sizeof(i) * 8)
+#define vod_number_of_residual_bits(i) ((i) % vod_number_of_bits(uint64_t))
+#define vod_array_length_for_bits(i) ((i) / vod_number_of_bits(uint64_t) + !!((i) % vod_number_of_bits(uint64_t)))
+#define vod_is_bit_set(mask, index) (!!((mask)[(index) / vod_number_of_bits((mask)[0])] & ((uint64_t)1 << ((index) % vod_number_of_bits((mask)[0])))))
+#define vod_set_bit(mask, index) ((mask)[(index) / vod_number_of_bits((mask)[0])] |= ((uint64_t)1 << ((index) % vod_number_of_bits((mask)[0]))))
+#define vod_reset_bit(mask, index) ((mask)[(index) / vod_number_of_bits((mask)[0])] &= ~((uint64_t)1 << ((index) % vod_number_of_bits((mask)[0]))))
+#define vod_set_bits(mask, max_bits) vod_memset((mask), 0xff, sizeof(uint64_t) * vod_array_length_for_bits(max_bits));
+#define vod_reset_bits(mask, max_bits) vod_memzero((mask), sizeof(uint64_t) * vod_array_length_for_bits(max_bits));
```

Comment:
True, these aren't used with the language mask or uint64_t*. Still, this might become an issue in the future, it would compile fine but be broken at runtime.

Are you willing to risk this?

Reply:
another option is create macros specifically for track_mask_t (in media_format.h) and use explicitly sizeof(track_mask_t)

Reply:
ok, so we would end up with 
```
vod_track_mask_set_bits
vod_track_mask_reset_bits
vod_track_mask_is_any_bit_set
```
etc...

Reply:
right, only the macros that care about the size.
it sounds ok to me, it's like we consider the track mask an object that has a set of methods associated with it